### PR TITLE
Add field selection info to the StandardSync persistence

### PR DESCRIPTION
**Pull request by mfsiega-airbyte**
## What
Include field selection in the standard sync persistence.

This is step 2 of 3 in backend changes to support new APIs for column selection.

Step 1 was https://github.com/airbytehq/airbyte/pull/20202 (database migration to add the column for this info) and Step 3 in the future will use this field to make a map from `StreamDescriptor` -> `true/false` whether column selection is enabled for that stream.

## How
- Add it to the model
- Make sure it gets written/read to/from the database

### CODE REVIEW

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml [hunk 1]
```diff
@@ -42,6 +42,11 @@ properties:
     type: string
   catalog:
     existingJavaType: io.airbyte.protocol.models.ConfiguredAirbyteCatalog
+  fieldSelectionData:
+    type: object
+    description: A map of stream description to an indicator of whether field selection is enabled for that stream.
```
Comment (alovew) [inline airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:47]: can you change stream description to StreamDescriptor? it's a special object for uniquely identifying streams.

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java [hunk 2]
```diff
@@ -187,6 +187,7 @@ private void writeStandardSync(final StandardSync standardSync, final DSLContext
           .set(CONNECTION.DESTINATION_ID, standardSync.getDestinationId())
           .set(CONNECTION.NAME, standardSync.getName())
           .set(CONNECTION.CATALOG, JSONB.valueOf(Jsons.serialize(standardSync.getCatalog())))
+          .set(CONNECTION.FIELD_SELECTION_DATA, JSONB.valueOf(Jsons.serialize(standardSync.getFieldSelectionData())))
```
Comment (alovew) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java:190]: will we need to add this for connection create as well? not sure of the UX here

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml (hunk 1 above)
Comment (mfsiega-airbyte) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml]: Done.

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java (hunk 2 above)
Comment (mfsiega-airbyte) [reply airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java:190]: Good call, not sure how I missed this - added.

**Review by alovew — APPROVED** [review]

**Review by pmossman — APPROVED** [review]
Nice, this looks good to me!

**Pull request merged**